### ISSUE TITLE
--inherit-env for packed actions

### ISSUE THREAD

**Issue by kfirstri** [StackStorm/st2#3888]
Hello,

I'm writing a new action that runs a bash file (runner type - local-shell-script)
my YAML file -
```yaml
---
name: create_ns
runner_type: local-shell-script
description: Create a new namespace
enabled: true
entry_point: 'shell/create_ns.sh'
parameters:
  name: 
    type: string
    position: 1
  cluster: 
    type: string
    default: 'dev'
    position: 2
  sudo:
    default: true
    immutable: true
  env:
    default:
      KUBECONFIG: "/path/to/file"
    immutable: true
```
In the stackstorm container i have a KUBECONFIG env variable, so what i'm really looking for is a way to pass environmant variable inside the execution.

i've seen in the documentation that --inherit-env passes the variables inside the execution, which is good, and if i use the CLI (`st2 run pack.create_ns -e) it works great! what i don't understand is how to specify it **inside** the YAML configuration?

Other methods i thought of - 
 - Passing the environment as a parameter - won't work because i can't use {{ env }} (yaml files are static)
 - Maybe chaining it somehow? not sure exactly with what..

**Comment by Kami** [comment]
`--inherit-env` is a CLI only option. What it basically does is it inherits all the environment variable which are available where the CLI command runs and passes them to an action using `env` parameter (nothing to do with the server side environment variables but the ones which are available where CLI command runs).

By default, Python and remote runners inherit and use all the environment variables which are available to the action runner process (process / service under which actions run).

So in short, you need to make sure that environment variable is set and available to the action runner services (you might need to tweak service manager / systemd configuration file to make sure variable is correctly set / passed through).

You can test faster / easier by using `st2 run core.local cmd='echo ${KUBECONFIG}'` command.

**Comment by kfirstri** [comment]
Thanks for the answer @Kami,

I'm running Stackstorm as a Nomad job, and using Consul for configuration (for example - path to KUBECONFIG files).

What I did eventually is used Nomad's template stanza to generate a file inside the stackstorm container, and copied the file into `/etc/default/st2actionrunner` which passes the environment variable inside each action runner.

**Comment by LindsayHill** [comment]
Setting it in `/etc/default/st2actionrunner`, as you did, is probably the best way to do this.
